libs/s25main/desktops/dskOptions.cpp

```diff
@@ -663,3 +685,89 @@ void dskOptions::loadVideoModes()
     // Sort by aspect ratio
     std::sort(video_modes.begin(), video_modes.end(), cmpVideoModes);
 }
+
+void dskOptions::Msg_ScreenResize(const ScreenResizeEvent& sr)
+{
+    Desktop::Msg_ScreenResize(sr);
+    updateGuiScale();
+}
+
+bool dskOptions::Msg_WheelUp(const MouseCoords& mc)
+{
+    if(VIDEODRIVER.GetModKeyState().ctrl)
+    {
+        scrollGuiScale(true);
+        return true;
+    } else
+        return Desktop::Msg_WheelUp(mc);
+}
+
+bool dskOptions::Msg_WheelDown(const MouseCoords& mc)
+{
+    if(VIDEODRIVER.GetModKeyState().ctrl)
+    {
+        scrollGuiScale(false);
+        return true;
+    } else
+        return Desktop::Msg_WheelDown(mc);
+}
+
+void dskOptions::updateGuiScale()
+{
+    // generate GUI scale percentages in 10% increments
+    constexpr auto stepSize = 10u;
+    const auto roundGuiScale = [=](unsigned percent) {
+        return helpers::iround<unsigned>(static_cast<float>(percent) / stepSize) * stepSize;
+    };
+
+    unsigned minPercent, maxPercent, recommendedPercent;
+    VIDEODRIVER.getGuiScaleRange(&minPercent, &maxPercent, &recommendedPercent);
+    const auto recommendedPercentRounded = roundGuiScale(recommendedPercent);
+    auto* combo = GetCtrl<ctrlGroup>(ID_grpGraphics)->GetCtrl<ctrlComboBox>(ID_cbGuiScale);
+
+    guiScales_.clear();
+    combo->DeleteAllItems();
+
+    guiScales_.push_back(0);
+    combo->AddString(helpers::format(_("Auto (%u%%)"), recommendedPercent));
+    if(SETTINGS.video.guiScale == 0)
+        combo->SetSelection(0);
+
+    for(unsigned percent = roundGuiScale(minPercent); percent <= maxPercent; percent += stepSize)
+    {
+        if(percent == recommendedPercentRounded)
+            recommendedGuiScaleIndex_ = guiScales_.size();
+        guiScales_.push_back(percent);
+
+        combo->AddString(helpers::toString(percent) + "%");
+        if(percent == SETTINGS.video.guiScale)
+            combo->SetSelection(combo->GetNumItems() - 1);
+    }
+
+    // if GUI scale exceeds maximum, lower it to keep UI elements on screen
+    if(SETTINGS.video.guiScale > guiScales_.back())
+    {
+        combo->SetSelection(combo->GetNumItems() - 1);
+        SETTINGS.video.guiScale = guiScales_.back();
+        VIDEODRIVER.setGuiScalePercent(SETTINGS.video.guiScale);
+    }
+}
+
+void dskOptions::scrollGuiScale(bool up)
+{
+    auto* group = GetCtrl<ctrlGroup>(ID_grpGraphics);
+    auto* combo = group->GetCtrl<ctrlComboBox>(ID_cbGuiScale);
+    const auto& selection = combo->GetSelection();
+    unsigned newSelection = 0;
+    if(!selection || *selection == 0) // No selection or "Auto" item selected
+        newSelection = recommendedGuiScaleIndex_ + 1;
```

Comment:
```suggestion
        newSelection = recommendedGuiScaleIndex_;
```

This is now a 1:1 match, isn't it?

Reply:
Indeed. I'll notice all these issues once I can actually launch and test it.